src/Microsoft.AspNet.Server.Kestrel/Http/Connection.cs

```diff
@@ -29,15 +30,17 @@ public Connection(ListenerContext context, UvStreamHandle socket) : base(context
             ConnectionControl = this;
 
             _connectionId = Interlocked.Increment(ref _lastConnectionId);
+
+            SocketInput = new SocketInput(Memory2);
+            SocketOutput = new SocketOutput(Thread, _socket, _connectionId, Log);
+            _frame = new Frame(this);
         }
 
         public void Start()
         {
             Log.ConnectionStart(_connectionId);
 
-            SocketInput = new SocketInput(Memory);
-            SocketOutput = new SocketOutput(Thread, _socket, _connectionId, Log);
-            _frame = new Frame(this);
+            Task.Run((Action)_frame.ProcessFraming);
```

Comment:
Remove cast and change ProcessFraming to Task returning; don't need to change function body